retry/retry.go

```diff
@@ -164,7 +164,8 @@ func BackoffNanos(
 	if retry >= 1 {
 		backoff = int64(float64(backoff) * math.Pow(backoffFactor, float64(retry-1)))
 	}
-	if jitter {
+	// Validate the value of backoff to make sure Int63n() does not panic.
+	if jitter && backoff >= 2 {
```

Comment:
good to have this check but maybe add a `Validate()` to `Options` and ensure the user provided numbers are valid too.

Reply:
I don't think the panic here is originated from invalid numbers in the options, the user could provide initialBackoff = 1 nano as a valid input and cause a panic in the original implementation.

Reply:
That's what i mean. If we're going to silently not jitter unless `initialBackoff` >= 2, feels like we should inform the user that the value they gave us was wrong. Cause they'd expect the retrier to jitter cause they'd specified it should.

Reply:
yeah even with initialBackoff == 1, say retry = 4, backoff will be  1 * 2^(4-1) = 8, which still can be jittered, so don't think initialBackoff is a blocker here, also that kind of value seems to be impractical in production, so I'd keep the interface simpler and save the validation call.